File: number_theory/primality_test.cpp
// primality_test.cpp - Check if a number is prime using trial division.
// Author: Scott

#include <iostream>
#include <cmath>
using namespace std;

bool is_prime(int n) {
    if (n <= 1) return false;
    for (int i = 2; i <= sqrt(n); ++i)
        if (n % i == 0) return false;
    return true;
}

int main() {
    int n = 29;
    cout << "Enter integer to test (default 29): ";
    string input;
    getline(cin, input);
    if (!input.empty()) n = stoi(input);

    if (is_prime(n))
        cout << n << " is prime." << endl;
    else
        cout << n << " is not prime." << endl;
    return 0;
}

File: number_theory/primality_test.py
"""
primality_test.py - Check if a number is prime using basic trial division.
Not efficient for large numbers but useful for educational purposes.
Author: Scott
"""

def is_prime(n):
    if n <= 1:
        return False
    if n <= 3:
        return True
    if n % 2 == 0 or n % 3 == 0:
        return False
    i = 5
    while i * i <= n:
        if n % i == 0 or n % (i + 2) == 0:
            return False
        i += 6
    return True

if __name__ == "__main__":
    n = int(input("Enter integer to check for primality: ") or "29")
    print(f"{n} is prime." if is_prime(n) else f"{n} is not prime.")
